src/dmd/cparse.d

```diff
@@ -445,7 +445,7 @@ final class CParser(AST) : Parser!AST
                     // not merged into the current case. This can happen for
                     // case 1: ... break;
                     // debug { case 2: ... }
-                    if (cur.isBreakStatement())
+                    if (cur && cur.isBreakStatement())
```

Comment:
Maybe this check should occur before `statements.push(cur);`?
Just in case further code tries to access the statements

Reply:
Right now the code depends on an array of statements with 1 null element to mean an empty case statement. Putting the check before pushing cur will change that.

Reply:
The rest of the compiler is generally quite resilient to this.

https://github.com/dlang/dmd/blob/ee27b0356052365ab6113eff3f8613228466bab7/src/dmd/statementsem.d#L361-L367

https://github.com/dlang/dmd/blob/ee27b0356052365ab6113eff3f8613228466bab7/src/dmd/s2ir.d#L784

Reply:
I see, thanks for the clarification